it/src/test/java/com/linecorp/centraldogma/it/WatchTest.java

```diff
@@ -455,6 +506,218 @@ void transformingThread_withCustom(ClientType clientType) {
+        await().untilAtomic(triggeredCount, Matchers.is(1));
+
+        final Revision rev0 = watcher.initialValueFuture().join().revision();
+
+        // change file
+        final Change<JsonNode> change1 = Change.ofJsonUpsert(
+                filePath, "{ \"a\": \"artichoke\"}");
+        final Revision rev1 = client.push(dogma.project(), dogma.repo1(), rev0, "Change /a", change1)
+                                    .join().revision();
+        await().untilAtomic(triggeredCount, Matchers.is(2));
+        assertThat(watchResult.get()).isEqualTo(watcher.latest());
+
+        // remove file
+        final Change<Void> change2 = Change.ofRemoval(filePath);
+        final Revision rev2 = client.push(dogma.project(), dogma.repo1(), rev1, "Removal", change2)
+                                    .join().revision();
+
+        // Wait over the timeoutMillis(100) + a
+        Thread.sleep(1000);
+
+        // check utilize latest data before removal
+        assertThat(triggeredCount.get()).isEqualTo(2);
+        assertThat(watchResult.get()).isEqualTo(watcher.latest());
+
+        // add file
+        final Change<JsonNode> change3 = Change.ofJsonUpsert(
+                filePath, "{ \"a\": \"apricot\", \"b\": \"banana\" }");
+        final Revision rev3 = client.push(dogma.project(), dogma.repo1(), rev2, "Add /a /b", change3)
+                                    .join().revision();
+        await().untilAtomic(triggeredCount, Matchers.is(3));
+        assertThat(watchResult.get()).isEqualTo(watcher.latest());
+    }
+
+    @ParameterizedTest
+    @EnumSource(ClientType.class)
+    void repositoryWatcher_entry_not_exist_error(
+            ClientType clientType) {
+        //Legacy client don't support this feature
+        if (clientType == ClientType.LEGACY) {
+            return;
+        }
+
+        revertTestFiles(clientType);
+
+        // prepare test
+        final CentralDogma client = clientType.client(dogma);
+        final String pathPattern = "/not_exist_repo/**";
+
+        // create watcher
+        final Watcher<Revision> watcher = client.repositoryWatcher(dogma.project(), dogma.repo1(),
+                                                                   pathPattern, Function.identity(),
+                                                                   blockingTaskExecutor, 100, true);
+
+        // check entry does not exist when to get initial value
+        assertThatThrownBy(watcher::awaitInitialValue).getRootCause().isInstanceOf(
+                EntryNotFoundException.class);
+
+        // when initialValueFuture throw 'EntryNotFoundException', you can't use 'watch' method.
+        assertThatThrownBy(() -> watcher.watch((rev, node) -> {
+        })).isInstanceOf(IllegalStateException.class);
+    }
+
+    @ParameterizedTest
+    @EnumSource(ClientType.class)
+    void
+    repositoryWatcher_entry_not_exist_error_and_watch_not_working_if_initialVal_completed_with_exception
```

Comment:
`repositoryWatcher_errorOnEntryNotFound` because this name is too long? 😆

Reply:
I changed the method name to be shorter.